src/Deferred.php

```diff
@@ -51,6 +53,11 @@ public function __construct(callable $waitCallback)
         $this->onRejectedCallbacks = [];
     }
 
+    /**
+     * @param callable(ResponseInterface): ResponseInterface|null $onFulfilled
```

Comment:
this should not be necessary with the `@implements` in the class docblock, it should come automatically from the covariant in the interface.